libethcore/CommonJS.cpp

```diff
@@ -22,16 +22,27 @@
  */
 
 #include "CommonJS.h"
+#include "ICAP.h"
 
 namespace dev
 {
 
-Address toAddress(std::string const& _sn)
+Address jsToAddress(std::string const& _s)
 {
-	if (_sn.size() == 40)
-		return Address(fromHex(_sn));
-	else
-		return Address();
+	try
+	{
+		eth::ICAP i = eth::ICAP::decoded(_s);
+		return i.direct();
+	}
+	catch (eth::InvalidICAP&) {}
+	try
+	{
+		auto b = fromHex(_s.substr(0, 2) == "0x" ? _s.substr(2) : _s, WhenError::Throw);
```

Comment:
- may be over zealous. if it is to be shown, it's an additional UI feature.
- 0x0000... is a perfectly valid address to be sending to - if it's being protected against, it's a UI feature.
- length checked in next line.